src/System/Params.cpp

```diff
@@ -0,0 +1,74 @@
+#include <System/Params.hpp>
+#include <JSystem/JSupport/JSUMemoryInputStream.hpp>
+
+bool TParams::load(const char* filename)
+{
+	bool found = false;
+	if (filename[0] == '/') {
+		filename++;
+	}
+
+	void* resource = nullptr;
+	if (mSceneArc != nullptr && mSceneArc->becomeCurrent("/map/params")) {
+		resource = mSceneArc->getResource(filename);
+	}
+
+	if (resource != nullptr) {
+		s32 size = mSceneArc->getResSize(resource);
+		JSUMemoryInputStream stream(resource, size);
```

Comment:
removing the `(void)!buffer;` inside of the stream's constructor fixes this, but breaks J2DSetScreen::J2DSetScreen

BUT! We can place the cheeky `(void)!res;` inside of J2DSetScreen's ctor and it'll still match! make this fix pls =)
what makes me think that this is correct and the assert was actually inside of J2DSetScreen is the fact that TP debug doesn't have any asserts inside of JSUMemoryInputStream and it is extremely unlikely that they would remove such an assert in a newer jsystem version

Reply:
Thanks!!